app/services/events/post_process_service.rb

```diff
@@ -95,8 +95,11 @@ def expire_cached_charges(subscriptions)
       end
     end
 
-    def flag_refresh_from_subscription
-      subscriptions.select(&:active?).each { |s| LifetimeUsages::FlagRefreshFromSubscriptionService.new(subscription: s).call }
+    def track_subscription_activity!
+      UsageMonitoring::TrackSubscriptionActivityService.call(
+        organization:,
+        subscription_ids: subscriptions.select(&:active?).pluck(:id)
```

Comment:
the `.pluck` here seems to indicate a DB query, (given there's a `select(&:active?)` it's not, so perhaps it's better to just write `map`? 🤔

Reply:
I have reworked the service and it takes a full model now